MyString.java

```diff
@@ -78,9 +120,26 @@ public static String randomStringOfLetters(int n) {
      * @return a string consisting of str1 minus all the characters of str2
      */
     public static String remove(String str1, String str2) {
-       //// Replace the following statement with your code
-        return null;
+        String a = "";
+        int i = 0;
+        while (i < str1.length()) {
+            int n = countChar(str2, str1.charAt(i));
+            if (n == 0) {
+                a = a +  str1.charAt(i);
+            }
+            else {
+            int index = str2.indexOf(str1.charAt(i));
+            if (index == 0) {
+                str2 = str2.substring(1, str2.length());
+            }
+            else {
+            str2 = str2.substring(0, index) + str2.substring(index +1, str2.length());

```

Comment:
else if.. 
and you dont need those 3 cases, see recitation 7 isAnagram question